cli/src/commands/auth.rs

```diff
@@ -89,32 +93,30 @@ pub async fn handle_auth(
         match handle_auth_register(config, config_path).await {
             Ok(_) => {
                 print_user_success!("{}", "User successfuly regsistered");
+                Ok(ExitCode::Ok.into())
             }
-            Err(error) => {
-                return Err(anyhow!(
-                    "User registration failed: {}",
-                    error.root_cause().to_string()
-                ))
-            }
+            Err(error) => Err(anyhow!(
+                "User registration failed: {}",
+                error.root_cause().to_string()
+            )),
         }
     } else if matches.subcommand_matches("login").is_some() {
         match handle_auth_login(config, config_path).await {
             Ok(_) => {
                 print_user_success!("{}", "User login successful");
+                Ok(ExitCode::Ok.into())
             }
-            Err(error) => {
-                return Err(anyhow!(
-                    "User login failed: {}",
-                    error.root_cause().to_string()
-                ));
-            }
+            Err(error) => Err(anyhow!(
+                "User login failed: {}",
+                error.root_cause().to_string()
+            )),
```

Comment:
This seems like another place that `ExitCode::AuthenticationFailure` would be appropriate

Reply:
Yeah, though I didn't want to impact the CLI's text output. Currently this does set an error code and I think it's better just to keep that error code and the text output consistent rather than switching for consistency with a mostly unrelated command.